### ISSUE TITLE
fog_key_pair fails with password protected ssh key

### ISSUE THREAD

**Issue by tatemz** [chef-boneyard/chef-provisioning-fog#179]
I kept getting errors while working with the DigitalOcean driver.

Here was my basic setup:

```
require "chef/provisioning"
require "chef/provisioning/fog_driver/recipe_dsl"

with_driver "fog:DigitalOcean"

fog_key_pair 'password_protected_rsa'

add_machine_options :bootstrap_options => {
  :image_distribution => "Ubuntu",
  :image_name => "14.04.4 x64",
  :flavor_name => "2GB",
  :region_name => "New York 2",
  :key_path => "/Users/tatemz/.ssh/password_protected_rsa"
}

machine 'chef' do
  recipe 'chef-server'
end
```

There were various errors with fog_key_pair from [this line](https://github.com/chef/chef-provisioning-fog/blob/master/lib/chef/provider/fog_key_pair.rb#L158) and [this line](https://github.com/chef/cheffish/blob/master/lib/cheffish/key_formatter.rb#L74).

I don't have a full debug output yet, as I just abandoned this method for a non-protected ssh key.

**Comment by jjasghar** [comment]
If i'm reading this this correctly just doesn't work, correct?

If so, I'm more then willing to put a note in the README about not using it, tbh, i don't think I ever have used it myself.

**Comment by tatemz** [comment]
Yeah. it gives an error specific to `key_type` not being an available type from cheffish.

**Comment by jjasghar** [comment]
awesome thanks. It would be nice to have support for this, but if this is just clarification in the README, could you put in a PR to fix the doc? :metal:

**Comment by nandanhegde73** [comment]
This issue or pull request is being automatically closed because the repository is being archived as part of the [Repository Standardization Initiative](https://github.com/chef-boneyard/oss-repo-standardization-2025).
